lazy_tensor_core/lazy_bench.py

```diff
@@ -501,6 +504,50 @@ def merge_with_prefix(prefix, tmp_dir, out_dir, headers):
         for l in results:
             acc_csv.write(l)
 
+def merge_reformat(tmp_dir, out_dir):
+
+    def get_field(row, name, file_type):
+        headers = {
+            "error": ("name", "test", "error"),
+            "lazy-compute" : ("name", "dev", "experiment", "test", "speedup", "pvalue"),
+            "lazy-overheads" : ("dev", "name", "test", "overhead", "pvalue") 
+        }
+
+        header = headers[file_type]
+        r = row[header.index(name)] if name in header else "N/A"
+        return r
+
+    table = collections.defaultdict(dict)
+
+    csv_files = glob.glob(os.path.join(tmp_dir, "*.csv"))
+    for csvf in csv_files:
+
+        with open(csvf, "r") as csvfile:
+            print(f"processing {csvf}")
+            prefix = os.path.basename(csvf).split("_")[0]
+            csvreader = csv.reader(csvfile)
+            # This skips the first row of the CSV file.
+            next(csvreader)
+
+            for r in csvreader:
+                key = (get_field(r, "name", prefix), get_field(r, "test", prefix))
+                entry = table[key]
+
+            if prefix == "error":
+                entry["error"] = get_field(r, "error", prefix)
+            elif prefix == "lazy-overheads":
+                entry["overhead"] = get_field(r, "overhead", prefix)
+            else:
+                entry[get_field(r, "experiment", prefix)] = get_field(r, "speedup", prefix)
+            
+
+    headers = ["name", "test", "amortized", "unamortized", "error"]
```

Comment:
it would be a little better to let the names of the experiments (amortized 10x, ...) come from the data itself, since we could change to 6x or whatever and don't want it to be hardcoded erroneously.  is this an easy change or should we fix that later?

Reply:
fixed!